src/feed-registry.ts

```diff
@@ -0,0 +1,33 @@
+import {
+  log,
+  Address,
+  Bytes,
+  BigInt,
+  BigDecimal,
+} from "@graphprotocol/graph-ts"
+
+import {
+  Token,
+  CollateralToken
+} from "../generated/schema"
+
+import {
+  FeedConfirmed
+} from "../generated/FeedRegistry/FeedRegistry"
+
+import {
+  getOrCreateToken,
+  getOrCreateCollateralToken
+} from "./utils/utils";
+
+export function handleCollateralToken(event: FeedConfirmed): void {
+  let contractAddress = event.params.asset.toHexString();
+
+  // Add tokens in Chainlink Feed Registry to Token entity
+  let token = getOrCreateToken(contractAddress);
+  token.save();
+
+  // Add tokens in Chainlink Feed Registry to CollateralToken entity
+  let collateralToken = getOrCreateCollateralToken(contractAddress);
```

Comment:
So instead of this, with oracle entity instead of CollatToken entity you'd do
`getOrCreateToken` then`oracle.supportTokens.push(token.id)` and oracle.save()

Reply:
`getOrCreateCollateralToken` token was deprecated and replaced by adding an optional parameter `isOracle` to the `getOrCreateToken` function in `utils` folder.